dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/ActivationWithLossLayer.java

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.layers;
+
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters.LayerIndex;
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters.NumberOfOutput;
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters.LossFunction;
+import edu.snu.dolphin.dnn.conf.NeuralNetworkConfigurationParameters.SerializedLayerConfiguartion;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tang.formats.ConfigurationSerializer;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+/**
+ * Loss Layer with activation function.
+ */
+public final class ActivationWithLossLayer extends LayerBase {
+
+  private final LayerBase activationLayer;
+  private final String lossFunction;
+
+  @Inject
+  private ActivationWithLossLayer(@Parameter(LayerIndex.class) final int index,
+                                  @Parameter(NumberOfOutput.class) final int numOutput,
+                                  @Parameter(SerializedLayerConfiguartion.class) final String serializedLayerConf,
+                                  final ConfigurationSerializer configurationSerializer,
+                                  @Parameter(LossFunction.class) final String lossFunction) {
+    super(index, numOutput);
+    this.lossFunction = lossFunction;
+
+    try {
+      final Configuration layerConf = Tang.Factory.getTang().newConfigurationBuilder(
+          configurationSerializer.fromString(serializedLayerConf))
+          .bindNamedParameter(LayerIndex.class, String.valueOf(index)) // bind a layer index for injecting inner layer.
+          .build();
+      this.activationLayer = Tang.Factory.getTang().newInjector(layerConf).getInstance(LayerBase.class);
+    } catch (final IOException ioException) {
+      throw new RuntimeException("IOException while de-serializing a layer configuration: " + ioException);
+    } catch (final InjectionException injectException) {
+      throw new RuntimeException("InjectionException while injecting activation layer: " + injectException);
+    }
+  }
+
+  @Override
+  public boolean isLearnable() {
+    return false;
+  }
+
+  @Override
+  public INDArray feedForward(final INDArray input) {
+    return activationLayer.feedForward(input);
+  }
+
+  /**
+   * Compute the error for the specified loss function.
+   * @param label the label value.
+   * @param activation the activation value.
+   * @param nextError an error of the next layer - this argument is ignored.
+   * @return the error with respect to the activation and label values.
+   */
+  @Override
+  public INDArray backPropagate(final INDArray label, final INDArray activation, final INDArray nextError) {
+    switch (lossFunction.toLowerCase()) {
+    case "cross-entropy":
```

Comment:
The other names don't use hyphens; are we sure we want to use this naming convention just for loss functions?